src/test/scala/nl/knaw/dans/easy/agreement/internal/PlaceholderMapperSpec.scala

```diff
@@ -200,233 +206,258 @@ class PlaceholderMapperSpec extends UnitSpec with MockFactory with BeforeAndAfte
-      )
-    }
+    testInstance.embargo(emd) should contain theSameElementsAs List(
+      (UnderEmbargo, false),
+      (DateAvailable, "1992-07-30")
+    )
   }
 
   it should "give the embargo keyword mappings with UnderEmbargo=false when no DateAvailable is available" in {
-    emd.getEmdDate _ expects () returning date
-    date.getEasAvailable _ expects () returning ju.Collections.emptyList()
-
-    testInstance.embargo(emd) should {
-      have size 2 and contain allOf (
-        (UnderEmbargo, false),
-        (DateAvailable, "")
-      )
-    }
+    emd.getEmdDate _ expects() returning date
+    date.getEasAvailable _ expects() returning ju.Collections.emptyList()
+
+    testInstance.embargo(emd) should contain theSameElementsAs List(
+      (UnderEmbargo, false),
+      (DateAvailable, "")
+    )
   }
 
   "metadataTable" should "give a mapping for the metadata elements in the dataset" in {
+    val ccBy = "https://creativecommons.org/licenses/by/4.0/legalcode"
+    expectEmd
+    expectLicenses(Seq("accept", ccBy))
+
+    testInstance.metadataTable(emd, Seq("abc", "def"), "datasetID:1234").asScala should contain theSameElementsAs List(
+      Map(MetadataKey.keyword -> "ghi", MetadataValue.keyword -> "item4<br/>item5<br/>item6").asJava,
+      Map(MetadataKey.keyword -> "def", MetadataValue.keyword -> "Anonymous").asJava,
+      Map(MetadataKey.keyword -> "abc", MetadataValue.keyword -> "abc; def").asJava,
+      Map(MetadataKey.keyword -> "license", MetadataValue.keyword -> ccBy).asJava,
+    )
+  }
+
+  it should "give a mapping with a cc0 license" in {
+    expectEmd
+    expectLicenses(Seq("accept"))
+    expectEmdRights(AccessCategory.OPEN_ACCESS)
+
+    testInstance.metadataTable(emd, Seq("abc", "def"), "datasetID:1234").asScala should contain
+      Map(
+        MetadataKey.keyword -> "license",
+        MetadataValue.keyword -> "http://creativecommons.org/publicdomain/zero/1.0/legalcode",
+      ).asJava
+  }
+
+  it should "give a mapping with a DANS license" in {
+    expectEmd
+    expectLicenses(Seq("accept"))
+    expectEmdRights(AccessCategory.GROUP_ACCESS)
+
+    testInstance.metadataTable(emd, Seq("abc", "def"), "datasetID:1234").asScala should contain
+      Map(
+        MetadataKey.keyword -> "license",
+        MetadataValue.keyword -> "http://dans.knaw.nl/en/about/organisation-and-policy/legal-information/DANSGeneralconditionsofuseUKDEF.pdf",
+      ).asJava
+  }
+
+  it should "map relations" in {
+    val term = new Term(Name.RELATION, Namespace.DCTERMS)
+    emd.getTerms _ expects() returning Set(term).asJava
+    val items: Seq[MetadataItem] = Seq(
+      new BasicString("foo"),
+      new Relation("bar"),
+      new Relation("rabarbera", new URI("http://xx.dans.knaw.nl/yy")),
+    )
+    emd.getTerm _ expects term returning items.asJava
+
+    testInstance.metadataTable(emd, Seq("abc", "def"), "datasetID:1234").asScala should contain theSameElementsAs Seq(
+      Map(
+        MetadataKey.keyword -> null,
```

Comment:
* I supose this should be something like "relations"
* I this broken by my refactoring?